artifactory/services/upload.go

```diff
@@ -138,7 +142,24 @@ func addSymlinkProps(artifact clientutils.Artifact, uploadParams UploadParams) (
 		}
 		artifactProps += utils.ARTIFACTORY_SYMLINK + "=" + artifactSymlink + sha1Property
 	}
-	artifactProps = addProps(uploadParams.GetProps(), artifactProps)
+	props := uploadParams.GetProps()
+	if strings.ContainsAny(props, "build.name") && strings.ContainsAny(props, "build.name") {
+		path, err := filepath.Abs(artifact.LocalPath)
+		if err != nil {
+			return "", err
+		}
+		revision, url, err := uploadParams.VcsData.GetVcsData(filepath.Dir(path))
+		if err != nil {
+			return "", err
+		}
+		if revision != "" {
+			props += ";build.revision=" + revision
+		}
+		if url != "" {
+			props += ";build.url=" + url
+		}
+	}
+	artifactProps = addProps(props, artifactProps)
```

Comment:
The above code should be here, because this function handles adding the sym link props.

Reply:
Moved